src/cache/redis.rs

```diff
@@ -0,0 +1,26 @@
+use crate::twin::Twin;
+
+use super::Cache;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use bb8_redis::{bb8::Pool, RedisConnectionManager};
+
+pub struct RedisCache {
+    pool: Pool<RedisConnectionManager>,
+}
+
+impl RedisCache {
+    pub async fn new(url: String) -> Result<Self> {
```

Comment:
I suggest that Redis cache takes the pool itself, not the url this way a single redis connection pool can be created i main, then be used by both storage and cache (or any other module that requires redis connection)

This will allow the entire app to share one connection pool instead of have many pools. it's a more efficient way to share connections